### ISSUE TITLE
Add support for a dataVariation

### ISSUE THREAD

**Issue by mcmurrym** [launchdarkly/ios-client#275]
Add times it would be nice if we could just get back a `Data` or <T: Decodable> representation of the value that is stored for a variation. This would help with `https://github.com/launchdarkly/ios-client-sdk/issues/274`.

Something like...
```
public func dataVariation<T: Decodable>(forKey flagKey: LDFlagKey, defaultValue: T) -> T {
    variationDetailInternal(flagKey, defaultValue, needsReason: false).value
}
public func dataVariationDetail<T: Decodable>(forKey flagKey: LDFlagKey, defaultValue: T) -> LDEvaluationDetail<T> {
    variationDetailInternal(flagKey, defaultValue, needsReason: true)
}
public func dataVariation(forKey flagKey: LDFlagKey, defaultValue: Data) -> Data {
    variationDetailInternal(flagKey, defaultValue, needsReason: false).value
}
public func dataVariationDetail(forKey flagKey: LDFlagKey, defaultValue: Data) -> LDEvaluationDetail<Data> {
    variationDetailInternal(flagKey, defaultValue, needsReason: true)
}
```
I would be more than happy to help create the solution, but because `LDValue` goes as deep as it does, it would take some non-insignificant changes.

**Comment by mkambale** [comment]
@mcmurrym I am badly struck in extracting the values. Please help me with this.
I am using LaunchDarkly 6.0 and getting flag value as a dictionary.
I am facing problem in reading the values from flag.
This is how I am getting flag value from server -

        let arrayValue = LDClient.get()!.jsonVariationDetail(forKey: forFlag, defaultValue: ["enabled": false])

And, I get this response -

<img width="438" alt="Screenshot 2022-07-15 at 2 44 36 PM" src="https://user-images.githubusercontent.com/28888982/179194822-739f613b-deb0-41cf-9e1b-005c8d5c3310.png">

My problem is how do I read the values from **arrayValue.value** ? I need to make certain changes depending on the values from dictionary. 
cc: @@louis-launchdarkly

**Comment by louis-launchdarkly** [comment]
Given this is a solution suggestion for the concern in #274, we will keep the discussion there for now.

**Comment by brianmichel** [comment]
I just ran into this issue as well, and it's a big bummer that LD does not make these values easily codable. It would be great if they were just parsed out as a regular set of struct as needed @louis-launchdarkly

**Comment by keelerm84** [comment]
We just released v9.4.0 which adds two new methods `variation<T>` and `variationDetail<T>`! Both of these methods accept a type that must be Decodable and implement our LDValueConvertible protocol (required for now to deal with some internal plumbing).
